x/ibc/04-channel/keeper/packet.go

```diff
@@ -326,6 +327,52 @@ func (k Keeper) ReceiveExecuted(
 	return nil
 }
 
+// WriteAcknowledgement writes the packet execution acknowledgement to the state,
+// which will be verified by the counterparty chain using AcknowledgePacket.
+//
+// CONTRACT: for synchronous execution, this function is be called in the IBC handler .
+// For async handling, it needs to be called directly by the module which originally
+// processed the packet.
+func (k Keeper) WriteAcknowledgement(
+	ctx sdk.Context,
+	packet exported.PacketI,
+	acknowledgement []byte,
+) error {
+	// NOTE: IBC app modules might have written the acknowledgement synchronously on
```

Comment:
I don't think we should do this, when would this function be called twice? That's a mistake by the module author, we don't want it to silently fail